File: 1957-delete-characters-to-make-fancy-string/1957-delete-characters-to-make-fancy-string.java
class Solution {
    public String makeFancyString(String s) {
        if (s.length() == 0) return s;  // Edge case for an empty string
        
        StringBuilder result = new StringBuilder();
        result.append(s.charAt(0)); // Append the first character
        int count = 1; // Start the count with 1 for the first character

        for (int i = 1; i < s.length(); i++) {
            if (s.charAt(i) == s.charAt(i - 1)) {
                count++; // Increment count if the current character is the same as the previous one
            } else {
                count = 1; // Reset count for a new character
            }
            
            // Append the current character if it does not appear more than twice consecutively
            if (count <= 2) {
                result.append(s.charAt(i));
            }
        }
        
        return result.toString(); // Convert StringBuilder to String and return
    }
}

File: 1957-delete-characters-to-make-fancy-string/1957-delete-characters-to-make-fancy-string.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        s = list(s)  # Convert string to list for in-place modifications
        length, i = 0, 0
        prev = '@'  # Placeholder to compare with the first character
        
        for c in s:
            if prev != c:  # Reset count if the character is new
                length = 1
            else:
                length += 1  # Increment count for consecutive characters
            
            # Only add to result if it's not more than two in a row
            if length <= 2:
                s[i] = c
                i += 1
            
            prev = c  # Update previous character
        
        return "".join(s[:i])  # Join only the relevant part of list as final string
